optimum/exporters/onnx/__main__.py

```diff
@@ -74,20 +75,17 @@ def main():
         task, args.model, framework=args.framework, cache_dir=args.cache_dir, trust_remote_code=args.trust_remote_code
     )
 
-    if task + "-with-past" in TasksManager.get_supported_tasks_for_model_type(
+    if task != "stable-diffusion" and task + "-with-past" in TasksManager.get_supported_tasks_for_model_type(
         model.config.model_type.replace("_", "-"), "onnx"
     ):
         if args.task == "auto":  # Make -with-past the default if --task was not explicitely specified
             task = task + "-with-past"
         else:
             logger.info(
-                f"The task `{args.task}` was manually specified, and past key values will not be reused in the decoding."
-                f"Please pass `--task {args.task}-with-past` to export using the past key values."
+                f"The task `{task}` was manually specified, and past key values will not be reused in the decoding."
+                f"Please pass `--task {task}-with-past` to export using the past key values."
```

Comment:
Same comment, why did it become `task`?

Reply:
`task` and `args.task` are equal in this case, I think it's clearer to only use `args.task` to verify whether the task detection was done automatically and to use `task` for the rest. I don't have a strong opinion on this so I can change it back if needed

Reply:
sounds good